### ISSUE TITLE
[Bug Report] `NumpyToTorch` wrapper ignores device upon reset

### ISSUE THREAD

**Issue by mantasu** [Farama-Foundation/Gymnasium#1106]
## Bug
[NumpyToTorch](https://github.com/Farama-Foundation/Gymnasium/blob/57136fbaeec340c023ae6939348ac2e7cfe3b3cb/gymnasium/wrappers/numpy_to_torch.py#L102) wrapper ignores `device` when the environment is reset. I created a PR that should fix it.

## Feature request
I would like to further ask if it was possible to change the wrapper's behavior to return tensorized `reward`, `truncated`, `terminated` for non-vectorized environments (or at least add an option to be able to conditionally return their tensor versions)? This is because it is more consistent and more convenient to handle such inputs, for example, when creating tensor buffers:

```python
for i, env_out in enumerate(env_outs[:-1]):
    buffer[i] = torch.cat([buffer[i], env_out.unsqueeze(0)])
```
I tried applying `vectors.numpy_to_torch.NumpyToTorch` wrapper on a non-vector environment which would do the trick, however, there is an assertion that requires the environment to be necessarily a vector env.

### Code example

```shell
import gymnasium as gym
from gymnasium.wrappers.numpy_to_torch import NumpyToTorch

env = NumpyToTorch(gym.make("CartPole-v1"), device="cuda")
print(env.reset()[0].device) # device(type='cpu')
```

### System info

`python=3.11`, `gymnasium=1.0.0a2`

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pseudo-rnd-thoughts** [comment]
Thanks for the issue and PR. 
Yes, I would have thought that all the data is updated to `torch.Tensor` including the reward, terminated and truncated data. 
Is that not happening?
